src/scheduler/scheduler.c

```diff
@@ -321,7 +321,7 @@ void initialize_worker_thread(void) {
 			type: INIT,
 			timestamp: 0.0,
 			send_time: 0.0,
-			mark: generate_mark(LidToGid(LPS_bound[t]->lid)),
+			mark: generate_mark(LPS_bound[t]->lid),
```

Comment:
I agree... the mark property is still using a global LPS ID. The fact is that the `generate_mark` function is expecting a local LPS ID cause internally is using the `LidToGid` function in order to convert it to a global one.